embassy-nrf/src/pwm.rs

```diff
@@ -447,17 +438,14 @@ impl<'d, T: Instance> SimplePwm<'d, T> {
 
         self.duty[channel] = duty & 0x7FFF;
 
-        r.seq0
-            .ptr
-            .write(|w| unsafe { w.bits(&self.duty as *const _ as u32) });
```

Comment:
you can't remove this write: self might have been moved from between it was created and now. You don't want it to use a stale pointer.